clang/include/clang/AST/GlobalDecl.h

```diff
@@ -164,7 +164,8 @@ class GlobalDecl {
   }
 
   static KernelReferenceKind getDefaultKernelReference(const FunctionDecl *D) {
-    return (D->hasAttr<DeviceKernelAttr>() || D->getLangOpts().CUDAIsDevice)
+    return (D->hasAttr<DeviceKernelAttr>() || D->getLangOpts().CUDAIsDevice ||
+            D->hasAttr<SYCLKernelAttr>())
```

Comment:
I'm not sure that checking for `SYCLKernelAttr` is correct here. We didn't check for it prior to https://github.com/llvm/llvm-project/pull/137882 landing.